// remx/dist/utils/testUtils.js
Object.defineProperty(exports,"__esModule",{value:true});exports.FlatList=exports.Text=exports.grabConsoleWarns=exports.grabConsoleErrors=exports.grabConsole=undefined;var _jsxFileName='src/utils/testUtils.js';var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _react=require('react');var _react2=_interopRequireDefault(_react);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}exports.default=

null;

var grabConsole=exports.grabConsole=function grabConsole(fn){var methods=arguments.length>1&&arguments[1]!==undefined?arguments[1]:['log','warn','error','trace'];
var originals={};
var calls=[];
methods.forEach(function(method){
originals[method]=console[method];
console[method]=function(){for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}
calls.push([method].concat(args));
};
});

fn();

methods.forEach(function(method){
console[method]=originals[method];
});

return calls;
};

var grabConsoleErrors=exports.grabConsoleErrors=function grabConsoleErrors(fn){return grabConsole(fn,['error']).map(function(args){return args.slice(1);});};
var grabConsoleWarns=exports.grabConsoleWarns=function grabConsoleWarns(fn){return grabConsole(fn,['warn']).map(function(args){return args.slice(1);});};var

Text=exports.Text=function(_React$Component){_inherits(Text,_React$Component);function Text(){_classCallCheck(this,Text);return _possibleConstructorReturn(this,(Text.__proto__||Object.getPrototypeOf(Text)).apply(this,arguments));}_createClass(Text,[{key:'render',value:function render()
{
return(
_react2.default.createElement('div',{__source:{fileName:_jsxFileName,lineNumber:30}},
this.props.children));


}}]);return Text;}(_react2.default.Component);var


FlatList=exports.FlatList=function(_React$PureComponent){_inherits(FlatList,_React$PureComponent);function FlatList(){_classCallCheck(this,FlatList);return _possibleConstructorReturn(this,(FlatList.__proto__||Object.getPrototypeOf(FlatList)).apply(this,arguments));}_createClass(FlatList,[{key:'render',value:function render()
{var _this3=this;
return this.props.data.map(function(item,i){
var Renderer=_this3.props.renderItem;
return _react2.default.createElement(Renderer,{item:item,key:_this3.props.keyExtractor(item,i),__source:{fileName:_jsxFileName,lineNumber:41}});
});
}}]);return FlatList;}(_react2.default.PureComponent);
